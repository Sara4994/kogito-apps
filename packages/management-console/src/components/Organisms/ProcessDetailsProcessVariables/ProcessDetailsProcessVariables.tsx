import { Card, CardBody, CardFooter, CardHeader, Text, TextContent, TextVariants } from '@patternfly/react-core';
import React from 'react';
import ReactJson from 'react-json-view';
import { Spinner } from '@patternfly/react-core/dist/esm/experimental';

const ProcessDetailsProcessVariables = ({ loading, data }) => {
  return (
    <Card style={{ overflowX: 'auto' }}>
      <CardHeader>Process Variables</CardHeader>
      <CardBody>
        <TextContent style={{ width: '30em' }}>
          {loading ? (
            data.ProcessInstances.map((item, index) => {
              return (
                <div key={index}>
                  <ReactJson src={JSON.parse(item.variables)}/>
                </div>
              );
            })
          ) : (
            <Spinner />
          )}
        </TextContent>
      </CardBody>
    </Card>
  );
};

export default ProcessDetailsProcessVariables;
