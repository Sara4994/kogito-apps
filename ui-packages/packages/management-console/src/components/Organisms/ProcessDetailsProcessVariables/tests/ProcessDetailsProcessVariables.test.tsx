import React from 'react';
import { shallow, mount } from 'enzyme';
import ProcessDetailsProcessVariables from '../ProcessDetailsProcessVariables';
import MockedReactJson from 'react-json-view';

jest.mock('react-json-view', () => jest.fn(() => null));
const props = {
  setUpdateJson: jest.fn(),
  displayLabel: false,
  updateJson: {
    trip: {
      begin: '2019-10-22T22:00:00Z[UTC]',
      city: 'Berlin',
      country: 'Germany',
      end: '2019-10-30T22:00:00Z[UTC]',
      visaRequired: false
    }
  },
  setDisplayLabel: jest.fn(),
  displaySuccess: false
};

const props2 = {
  setUpdateJson: jest.fn(),
  displayLabel: true,
  updateJson: {
    trip: {
      begin: '2019-10-22T22:00:00Z[UTC]',
      city: 'Berlin',
      country: 'Germany',
      end: '2019-10-30T22:00:00Z[UTC]',
      visaRequired: false
    }
  },
  setDisplayLabel: jest.fn(),
  displaySuccess: true
};
describe('ProcessVariables component tests', () => {
  it('snapshot testing without variables', () => {
    const wrapper = shallow(<ProcessDetailsProcessVariables {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('snapshot testing with variables', () => {
    const wrapper = mount(<ProcessDetailsProcessVariables {...props2} />);
    expect(wrapper).toMatchSnapshot();
    const onEdit = () => {
      return null;
    };
    const obj = {
      name: false,
      onEdit,
      src: {
        trip: {
          begin: '2019-10-22T22:00:00Z[UTC]',
          city: 'Berlin',
          country: 'Germany',
          end: '2019-10-30T22:00:00Z[UTC]',
          visaRequired: false
        }
      }
    };
    // expect(MockedReactJson).toHaveBeenCalledWith({...obj})
  });
});
