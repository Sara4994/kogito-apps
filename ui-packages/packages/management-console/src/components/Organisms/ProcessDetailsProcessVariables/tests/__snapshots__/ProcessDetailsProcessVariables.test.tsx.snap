// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessVariables component tests snapshot testing with variables 1`] = `
<Memo(ProcessDetailsProcessVariables)
  displayLabel={true}
  displaySuccess={true}
  setDisplayLabel={[MockFunction]}
  setUpdateJson={[MockFunction]}
  updateJson={
    Object {
      "trip": Object {
        "begin": "2019-10-22T22:00:00Z[UTC]",
        "city": "Berlin",
        "country": "Germany",
        "end": "2019-10-30T22:00:00Z[UTC]",
        "visaRequired": false,
      },
    }
  }
>
  <Card>
    <article
      className="pf-c-card"
      data-ouia-component-id={0}
      data-ouia-component-type="PF4/Card"
      data-ouia-safe={true}
    >
      <CardHeader>
        <div
          className="pf-c-card__header"
        >
          <Title
            headingLevel="h3"
            size="xl"
          >
            <h3
              className="pf-c-title pf-m-xl"
            >
              Process Variables
            </h3>
          </Title>
          <Label
            color="orange"
            icon={
              <InfoCircleIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
          >
            <span
              className="pf-c-label pf-m-orange"
            >
              <span
                className="pf-c-label__content"
              >
                <span
                  className="pf-c-label__icon"
                >
                  <InfoCircleIcon
                    color="currentColor"
                    noVerticalAlign={false}
                    size="sm"
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                        transform=""
                      />
                    </svg>
                  </InfoCircleIcon>
                </span>
                 
                Changes are not saved yet
              </span>
            </span>
          </Label>
          <Label
            className="kogito-management-console--variables__label-fadeIn"
            color="green"
            icon={
              <InfoCircleIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
          >
            <span
              className="pf-c-label pf-m-green kogito-management-console--variables__label-fadeIn"
            >
              <span
                className="pf-c-label__content"
              >
                <span
                  className="pf-c-label__icon"
                >
                  <InfoCircleIcon
                    color="currentColor"
                    noVerticalAlign={false}
                    size="sm"
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                        transform=""
                      />
                    </svg>
                  </InfoCircleIcon>
                </span>
                 
                Changes are saved
              </span>
            </span>
          </Label>
        </div>
      </CardHeader>
      <CardBody>
        <div
          className="pf-c-card__body"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <div>
                <mockConstructor
                  name={false}
                  onEdit={[Function]}
                  src={
                    Object {
                      "trip": Object {
                        "begin": "2019-10-22T22:00:00Z[UTC]",
                        "city": "Berlin",
                        "country": "Germany",
                        "end": "2019-10-30T22:00:00Z[UTC]",
                        "visaRequired": false,
                      },
                    }
                  }
                />
              </div>
            </div>
          </TextContent>
        </div>
      </CardBody>
    </article>
  </Card>
</Memo(ProcessDetailsProcessVariables)>
`;

exports[`ProcessVariables component tests snapshot testing without variables 1`] = `
<Card>
  <CardHeader>
    <Title
      headingLevel="h3"
      size="xl"
    >
      Process Variables
    </Title>
    <Label
      className="kogito-management-console--variables__label-fadeOut"
      color="green"
      icon={
        <InfoCircleIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      }
    >
       
      Changes are saved
    </Label>
  </CardHeader>
  <CardBody>
    <TextContent>
      <div>
        <mockConstructor
          name={false}
          onEdit={[Function]}
          src={
            Object {
              "trip": Object {
                "begin": "2019-10-22T22:00:00Z[UTC]",
                "city": "Berlin",
                "country": "Germany",
                "end": "2019-10-30T22:00:00Z[UTC]",
                "visaRequired": false,
              },
            }
          }
        />
      </div>
    </TextContent>
  </CardBody>
</Card>
`;
